waiter/src/waiter/process_request.clj

```diff
@@ -134,46 +134,41 @@
    `request-state-chan` is closed. Takes `instance-rpc-chan`, `service-id` and
    `reason-map` to acquire the instance.
-   `ex-handler` is called if an exception occurs while trying to acquire the instance.
    If an exception has occurred, no instance was acquired.
 
    Returns the instance if it was acquired successfully, otherwise,
    will return the output of `ex-handler`"
-  [instance-rpc-chan service-id {:keys [request-id] :as reason-map} start-new-service-fn request-state-chan ex-handler
+  [instance-rpc-chan service-id {:keys [request-id] :as reason-map} start-new-service-fn request-state-chan
    queue-timeout-ms reservation-status-promise metric-group add-debug-header-into-response!]
-  (async/go
-    (try
-      (log/debug "retrieving instance for" service-id "using" (dissoc reason-map :cid :time))
-      (add-debug-header-into-response! "X-Waiter-Service-Id" service-id)
-      (let [correlation-id (cid/get-correlation-id)
-            instance (<? (service/get-available-instance
-                           instance-rpc-chan service-id reason-map start-new-service-fn queue-timeout-ms metric-group add-debug-header-into-response!))]
-        (au/on-chan-close request-state-chan
-                          (fn on-request-state-chan-close []
-                            (cid/cdebug correlation-id "request-state-chan closed")
-                            ; request potentially completing normally if no previous status delivered
-                            (deliver reservation-status-promise :success)
-                            (let [status (deref reservation-status-promise 100 :unrealized-promise)] ; don't wait forever
-                              (cid/cinfo correlation-id "done processing request" status)
-                              (when (= :success status)
-                                (counters/inc! (metrics/service-counter service-id "request-counts" "successful")))
-                              (when (= :generic-error status)
-                                (cid/cerror correlation-id "there was a generic error in processing the request;"
-                                            "if this is a client or server related issue, the code needs to be updated."))
-                              (when (= :unrealized-promise status)
-                                (cid/cerror correlation-id "reservation status not received!"))
-                              (when (not= :success-async status)
-                                (counters/dec! (metrics/service-counter service-id "request-counts" "outstanding"))
-                                (statsd/gauge-delta! metric-group "request_outstanding" -1))
-                              (service/release-instance-go instance-rpc-chan instance {:status status, :cid correlation-id, :request-id request-id})))
-                          (fn [e] (log/error e "error releasing instance!")))
-        instance)
-      (catch Exception e
-        (ex-handler e)))))
+  (go-try
```

Comment:
Removed `ex-handler`.  Combined `(go (try ..))` into `(go-try ..)`  Code for `ex-handler` moved to `catch` block that calls `prepare-instance`